<launch>
  <!-- Arguments -->
  <arg name="model" default="robot" />
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>
  <arg name="local_planner_method" default="dwa_local_planner" doc="teb_local_planner base_local_planner"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <!-- dwa -->
    <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> -->
    <!-- <rosparam file="$(find ifly_navigation)/param/dwa_local_planner_params.yaml" command="load" />  -->

    <!-- teb -->
    <!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" /> -->
    <!-- <rosparam file="$(find ifly_navigation)/param/teb_local_planner_params.yaml" command="load" />  -->
    
    <!-- eband -->
    <!-- <param name="base_local_planner" value="eband_local_planner/EBandPlannerROS"/> -->
    <!-- <rosparam file="$(find ifly_navigation)/param/eband_local_planner_params.yaml" command="load" />  -->

    <param name="base_global_planner" value="global_planner/GlobalPlanner"/>

    <rosparam file="$(find ifly_navigation)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ifly_navigation)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ifly_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ifly_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find ifly_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find ifly_navigation)/param/global_planner_params.yaml" command="load" />

    

    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
</launch>
